timelock-impl/src/main/java/com/palantir/atlasdb/timelock/adjudicate/TimeLockClientFeedbackResource.java

```diff
@@ -16,35 +16,42 @@
 
 package com.palantir.atlasdb.timelock.adjudicate;
 
+import java.util.function.Function;
+
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.palantir.atlasdb.timelock.adjudicate.feedback.TimeLockClientFeedbackService;
 import com.palantir.atlasdb.timelock.adjudicate.feedback.TimeLockClientFeedbackServiceEndpoints;
 import com.palantir.atlasdb.timelock.adjudicate.feedback.UndertowTimeLockClientFeedbackService;
 import com.palantir.conjure.java.undertow.lib.UndertowService;
+import com.palantir.paxos.Client;
 import com.palantir.timelock.feedback.ConjureTimeLockClientFeedback;
 import com.palantir.tokens.auth.AuthHeader;
 
 public class TimeLockClientFeedbackResource implements UndertowTimeLockClientFeedbackService {
+    private Function<Client, Boolean> leadershipCheck;
```

Comment:
nit: `Predicate<Client>` avoids boxing/generics overheads, and I'd argue is more descriptive here